### PR TITLE
SONARJAVA-1251 S1228 Package-Info raise false positive when analyzing a single file

### PR DESCRIPTION
**Pull request by mpaladin**
https://jira.sonarsource.com/browse/SONARJAVA-1251

### CODE REVIEW

**Comment by mpaladin** [comment]
@benzonico code removed from `ChecksBridge` and added the ability to report issues on a file different from the one analysed, feedback is welcome as I don't fully like it.

File: java-squid/src/main/java/org/sonar/java/SonarComponents.java [hunk 1]
```diff
@@ -98,6 +108,10 @@ public Highlightable highlightableFor(File file) {
     return resourcePerspectives.as(Highlightable.class, inputFromIOFile(file));
   }
 
+  public Issuable issuableFor(File file) {
```
Comment (benzonico) [inline java-squid/src/main/java/org/sonar/java/SonarComponents.java]: Test ?

File: java-squid/src/main/java/org/sonar/java/model/VisitorsBridge.java [hunk 2]
```diff
@@ -242,6 +250,37 @@ public String getFileKey() {
       return sourceFile.getKey();
     }
 
+    @CheckForNull
+    private RuleKey getRuleKey(JavaCheck check) {
```
Comment (benzonico) [inline java-squid/src/main/java/org/sonar/java/model/VisitorsBridge.java]: This is not related to this ticket, but doing this should probably lead to remove completely the ChecksBridge class and report directly from the context class. If you share the feeling can you please create the ticket ?

Comment (mpaladin) [inline java-squid/src/main/java/org/sonar/java/model/VisitorsBridge.java]: @benzonico I fully share, right now the index is per project, it should be at least per file or on the go, ticket created: https://jira.sonarsource.com/browse/SONARJAVA-1256

File: java-squid/src/main/java/org/sonar/java/SonarComponents.java (hunk 1 above)
Comment (mpaladin) [inline java-squid/src/main/java/org/sonar/java/SonarComponents.java]: @benzonico test would not be realistic with `ResourcePerspective`.

**Pull request merged**